packages/cli/lib/services/parser.service.ts

```diff
@@ -70,15 +71,20 @@ class ParserService {
             'triggerAction'
         ];
 
-        const disallowedActionCalls = ['batchSend', 'batchSave', 'batchDelete'];
+        const disallowedActionCalls = ['batchSend', 'batchSave', 'batchDelete', 'batchUpdate'];
 
         const deprecatedCalls: Record<string, string> = {
             batchSend: 'batchSave',
             getFieldMapping: 'getMetadata',
             setFieldMapping: 'setMetadata'
         };
 
-        const callsReferencingModelsToCheck = ['batchSave', 'batchDelete'];
+        const callsProxy = ['proxy', 'get', 'post', 'put', 'patch', 'delete'];
+        const callsBatchingRecords = ['batchSave', 'batchDelete', 'batchUpdate'];
+        const callsReferencingModelsToCheck = callsBatchingRecords.concat('setMergingStrategy');
+        const proxyLines: number[] = [];
+        const batchingRecordsLines: number[] = [];
+        const setMergingStrategyLines: number[] = [];
```

Comment:
see below. addressing footguns by raising compilation errors and enforce a couple of rules:
- setMergingStrategy must be called only once
- setMergingStrategy must be called before any batch operation
- setMergingStrategy must be called before any proxy/get/post/...

Reply:
very nice